src/main/java/com/spotify/reaper/resources/CommonTools.java

```diff
@@ -133,78 +142,47 @@ public static RepairRun registerRepairRun(AppContext context, Cluster cluster,
    * backend.
    */
-  private static RepairRun storeNewRepairSegments(AppContext context, List<RingRange> tokenSegments,
-                                             RepairRun repairRun, RepairUnit repairUnit) {
+  private static List<RepairSegment.Builder> createRepairSegments(List<RingRange> tokenSegments, RepairUnit repairUnit){
+
     List<RepairSegment.Builder> repairSegmentBuilders = Lists.newArrayList();
-    for (RingRange range : tokenSegments) {
-      RepairSegment.Builder repairSegment = new RepairSegment.Builder(repairRun.getId(), range,
-                                                                      repairUnit.getId());
-      repairSegmentBuilders.add(repairSegment);
-    }
-    context.storage.addRepairSegments(repairSegmentBuilders, repairRun.getId());
-    if (repairRun.getSegmentCount() != tokenSegments.size()) {
-      LOG.debug("created segment amount differs from expected default {} != {}",
-                repairRun.getSegmentCount(), tokenSegments.size());
-      RepairRun newRepairRun = repairRun.with().segmentCount(tokenSegments.size()).build(repairRun.getId());
-      context.storage.updateRepairRun(newRepairRun);
-      
-      return newRepairRun;
-    }
-    
-    return repairRun;
+    tokenSegments.forEach((range) -> repairSegmentBuilders.add(new RepairSegment.Builder(range, repairUnit.getId())));
+    return repairSegmentBuilders;
   }
   
   
   /**
    * Creates the repair runs linked to given RepairRun and stores them directly in the storage
    * backend in case of incrementalRepair
    */
-  private static RepairRun storeNewRepairSegmentsForIncrementalRepair(AppContext context, Map<String, RingRange> nodes,
-                                             RepairRun repairRun, RepairUnit repairUnit) {
+  private static List<RepairSegment.Builder> createRepairSegmentsForIncrementalRepair(
+          Map<String, RingRange> nodes,
+          RepairUnit repairUnit) {
+
     List<RepairSegment.Builder> repairSegmentBuilders = Lists.newArrayList();
-    for (Entry<String, RingRange> range : nodes.entrySet()) {
-      RepairSegment.Builder repairSegment = new RepairSegment.Builder(repairRun.getId(), range.getValue(),
-                                                                      repairUnit.getId());
-      repairSegment.coordinatorHost(range.getKey());
-      repairSegmentBuilders.add(repairSegment);
-    }
-    context.storage.addRepairSegments(repairSegmentBuilders, repairRun.getId());
-    if (repairRun.getSegmentCount() != nodes.keySet().size()) {
-      LOG.debug("created segment amount differs from expected default {} != {}",
-                repairRun.getSegmentCount(), nodes.keySet().size());
-      RepairRun newRepairRun = repairRun.with().segmentCount(nodes.keySet().size()).build(repairRun.getId());
-      context.storage.updateRepairRun(newRepairRun);
-      
-      return newRepairRun;
-    }
-    
-    return repairRun;
+
+    nodes.entrySet().forEach((range)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "range" parameter [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1611)